core/capabilities/remote/dispatcher.go

```diff
@@ -128,7 +180,19 @@ func (d *dispatcher) receive() {
 				d.tryRespondWithError(msg.Sender, body, types.Error_CAPABILITY_NOT_FOUND)
 				continue
 			}
-			receiver.Receive(body)
+
+			// Calculate the usage of the receiver channel and set the gauge
+			// ??  Both operations are relatively cheap and in the context of capability message processing the cost should not
+			// be of concern ??
+			receiverQueueUsage := float64(len(receiver.ch)) / receiverBufferSize
+			capReceiveChannelUsage.WithLabelValues(k.capId, k.donId).Set(receiverQueueUsage)
+			select {
+			case receiver.ch <- body:
+			default:
+				// ?? Drop the message to stop blocking the receive loop, or block??
+				// ?? If dropping rely on the metrics to indicate issues, could also log a warning but would need
+				// some kind of rate limiting to avoid spamming the logs, arguably not necessary if you have metrics ??
```

Comment:
I'd rather spam logs than drop silently. We can deal with log rate limiting later.

Reply:
fair enough - I've add that in - left the comments for now until second review done

Reply:
@ettec Why drop it immediately vs waiting some minimal amount of time?

Reply:
if its being dropped, that implies the queue is full,  which has a buffer size of 10k, so its safe to assume if it gets to 10k queue backlog the capability is hosed and theres no point waiting

Reply:
the queue monitoring will highlight if there are issues way before that point in practice